doc/examples/applications/plot_cornea_spot_inpainting.py

```diff
@@ -0,0 +1,255 @@
+plot_comparison(mask_2, mask_open, "mask before", "after opening")
+
+#####################################################################
+# Next, we can make the detected areas wider by applying a dilation filter:
+
+mask_dilate = morphology.dilation(mask_open, footprint)
+plot_comparison(mask_open, mask_dilate, "before", "after dilation")
+
+#####################################################################
+# Dilation enlarges bright regions and shrinks dark regions.
+# Notice how, indeed, the white spots have thickened.
+
+#####################################################################
+# Inpaint each frame separately
+# =============================
+# We are now ready to apply inpainting to each frame. For this we use function
+# ``inpaint_biharmonic`` from the ``restoration`` module. It implements an
+# algorithm based on biharmonic equations.
+# This function takes two arrays as inputs:
+# The image to restore and a mask (with same shape) corresponding to the
+# regions we want to inpaint.
+
+image_seq_inpainted = np.zeros(image_seq.shape)
+
+for i in range(image_seq.shape[0]):
+    image_seq_inpainted[i] = restoration.inpaint_biharmonic(
+        image_seq[i],
+        mask_dilate
+    )
+
+#####################################################################
+# Let us visualize one restored image, where the dark spots have been
+# inpainted. First, we find the contours of the dark spots (i.e., of the mask)
+# so we can draw them on top of the restored image:
+
+contours = measure.find_contours(mask_dilate)
+
+# Gather all (row, column) coordinates of the contours
+x = []
+y = []
+for contour in contours:
+    x.append(contour[:, 0])
+    y.append(contour[:, 1])
+# Flatten them
+x_flat = np.concatenate(x).ravel()
+y_flat = np.concatenate(y).ravel()
+# Create a sparse matrix corresponding to the mask of these contours
+data = np.ones(x_flat.shape, dtype='bool')
+mtx = sparse.coo_matrix((data, (x_flat, y_flat)), shape=mask_dilate.shape)
+# Convert it to array
+arr = mtx.toarray().astype('int')
+# Pick one frame
+sample_result = image_seq_inpainted[12]
+# Normalize it (so intensity values range [0, 1])
+sample_result /= sample_result.max()
+
+#####################################################################
+# We use function ``label2rgb`` from the ``color`` module to overlay the
+# restored image with the segmented spots, using transparency (alpha
+# parameter).
+
+color_contours = color.label2rgb(
+    arr,
+    image=sample_result,
+    alpha=0.4,
+    bg_label=0
+)
+
+fig, ax = plt.subplots(figsize=(6, 6))
+
+ax.imshow(color_contours)
+ax.set_title('Segmented spots over restored image')
+
+fig.tight_layout()
+
+#####################################################################
+# Note that the dark spot located at (x, y) ~ (719, 1237) stands out; ideally,
+# it should have been segmented and inpainted. We can see that we 'lost' it to
+# the opening processing step, when removing fine-grained features.
```

Comment:
@decorouz @ana42742 @lagru should we try and improve our segmentation? Or leave it as an implicit challenge for the reader? :joy:

Reply:
Yes, we should. 
I observe we are applying `morphology.opening` operation twice. Any particular reason for that?
By applying the opening operation one time, it improves the segmentation and inpainting outputs.

Reply:
My 2cts: as along as the example uses best practices and a solid sensible approach I'm totally fine with leaving room for optimization. 

@decorouz, the opening is applied with differently sized footprints which can make a significant difference depending on the spot size. I think the larger the footprint the larger the spots that are removed. Might be a good explanation to put in there.  

> By applying the opening operation one time, it improves the segmentation and inpainting outputs.

Why do you think that the segmentation is improved?

Reply:
> My 2cts: as along as the example uses best practices and a solid sensible approach I'm totally fine with leaving room for optimization.

I agree with @lagru. There is not a *single way* of segmenting... anything. That being said, @decorouz if you would like to suggest an alternative workflow, please do so [here](https://github.com/mkcor/draft-notebooks/tree/main/inpaint).

> I think the larger the footprint the larger the spots that are removed. Might be a good explanation to put in there.

Ok. I think it shows pretty well on the figures, but let's be verbose and explicit about it.

> > I observe we are applying `morphology.opening` operation twice.

@decorouz it is not applied twice in the processing pipeline. It's done first with a size-1 footprint, then with a size-3 footprint ("Let us try now with a larger footprint:"), and the latter result is kept in the next processing steps.

Reply:
I really like that suggestion and think having a clear and linear workflow beats applying opening twice. 

> possibly inviting the reader to try out other parameter values on their own (in the same spirit as @ana42742's awesome note).

:+1:

Reply:
@decorouz would you be able to submit your suggestions? Thanks!

Reply:
Yes, I submitted my implementation [here](https://github.com/mkcor/draft-notebooks/tree/main/inpaint).

Reply:
Thanks, @decorouz. In your [notebook](https://github.com/mkcor/draft-notebooks/blob/7bee19482f5ad3757947c9cdfba870da06f8b580/inpaint/biola_inpaint.ipynb), I can see the removal of one *opening* operation (the one using `footprint = morphology.diamond(3)`, surprisingly), but the accompanying text isn't updated/adapted at all (so it doesn't even match at this point)... Editing the *narrative* part of the tutorial *is* the *work* ("narrative documentation").

Reply:
@decorouz for example: 75a1d273f748e9537ae579c8e3f4003b5219f9e1

Reply:
You are right, I will update the narrative as appropriate.  I suppose there are other parts of the notebook that needs updated

Reply:
>  I suppose there are other parts of the notebook that needs reflect the commits.

Not sure what you mean here. Did you suggest other changes to the data analysis workflow?

Reply:
No. I mean in this [notebook](https://github.com/mkcor/draft-notebooks/blob/7bee19482f5ad3757947c9cdfba870da06f8b580/inpaint/biola_inpaint.ipynb). Besides [this](https://github.com/scikit-image/scikit-image/pull/6853#discussion_r1234277296), there maybe other narrative parts of the notebook the should reflect my implementation.

Reply:
@decorouz I'm sorry, I'm not following here (*you* know whether *you* have made other changes, right? What's the "maybe?"). Let us take this conversation over to Zulip to straighten this out and keep noise down. Thanks!